website/docs/intro-to-webassembly/why-webassembly.md

```diff
@@ -1 +1,13 @@
-# Why WebAssembly
\ No newline at end of file
+# Why WebAssembly
+
+As the web has progressed from document database to application platform to runs-damn-near-everything platform, the needs of the web browser evolved. While JavaScript is a vibrant and powerful language with a wide variety of libraries and a robust ecosystem, many computing platforms rely on other languages.
+
+When building high-performance applications, some programming languages lend themselves better to the task of processing large amounts of data, running machine learning models, rendering complex graphics, etc. The advent of WebAssembly came from the desire to run C++ code natively in the browser. After an inital experiment called asm.js which transpiled C++ to JavaScript, a successor was designed to execute bytecode in a sandboxed environment. This was the evolution of WebAssembly into the portable, performant, and flexible technology we know today.
```

Comment:
`asm.js`, similar to wasm, was just a compile target—Emscripten was the project that compiled C++ to JS!

As an aside, I _really_ dislike the term "transpile"; a compiler is a compiler.

Reply:
Also, asm.js was sandboxed just because it was JS (not as strong as wasm, but still a sandbox). Wasm was simply the evolution of asm.js built from all of the learnings 🤘

Reply:
I updated the wording, check it out